src/commands/dev/dev.js

```diff
@@ -291,6 +293,24 @@ const dev = async (options, command) => {
   process.env.URL = url
   process.env.DEPLOY_URL = url
 
+  const startNetligraphWatcher = Boolean(options.graph)
+
+  if (startNetligraphWatcher && options.offline) {
+    console.warn(`${NETLIFYDEVERR} Warning: unable to start Netlify Graph in offline mode`)
+  } else if (startNetligraphWatcher && !site.id) {
+    console.warn(
+      `${NETLIFYDEVERR} Warning: no siteId defined, unable to start Netlify Graph. To enable, run ${chalk.yellow(
+        'netlify init',
+      )} or ${chalk.yellow('netlify link')}?`,
+    )
+    process.exit(1)
+  } else if (startNetligraphWatcher) {
```

Comment:
no need to else if if you use a `process.exit`.

But I would use the `error` function instead the `warn` when you want to exit. In general try to avoid process exit as we cannot do telemetry stuff afterwards. try to return or use our methods that end the command.


```javascript
if (startNetligraphWatcher && !site.id) {
  error(`No site id defined, unable to start Netlify Graph.
To enable it, run ${chalk.yellow('netlify init')} or ${chalk.yellow('netlify link')}`)
}
```

In general if you exit the process because something was not defined please use an error and try to phrase it as a statement and not a question.